sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -77,11 +95,29 @@ public static ReactorNettyClient createWithConnectionProvider(ConnectionProvider
             .observe(getConnectionObserver())
             .resolver(DefaultAddressResolverGroup.INSTANCE);
         reactorNettyClient.configureChannelPipelineHandlers();
-        //  This enables fast warm up of HttpClient
-        reactorNettyClient.httpClient.warmup().block();
+        attemptToWarmupHttpClient(reactorNettyClient);
         return reactorNettyClient;
     }
 
+    /*
+     * This enables fast warm up of HttpClient
+     */
+    private static void attemptToWarmupHttpClient(ReactorNettyClient reactorNettyClient) {
+        // Warmup wasn't found, so don't attempt it.
+        if (HTTP_CLIENT_WARMUP == null) {
+            return;
+        }
+
+        try {
+            ((Mono<?>) HTTP_CLIENT_WARMUP.invoke(reactorNettyClient.httpClient)).block();
+        } catch (ClassCastException | WrongMethodTypeException throwable) {
+            // Invocation failed.
```

Comment:
Should we log this (i.e. something we want to investigate when log appear) or fine with ignoring these exceptions?

Reply:
I think we should, but not on info level, may be debug level logging.

Reply:
I would go with `debug` as it won't affect runtime of an application, aside not having a warm startup.

Reply:
Makes sense, thanks @alzimmermsft